circuits/cpp/barretenberg/cpp/src/barretenberg/stdlib/primitives/field/field.test.cpp

```diff
@@ -120,74 +127,92 @@ template <typename Composer> class stdlib_field : public testing::Test {
+                     * not from a copy constraint. That failure is because the assert_equal
+                     * below says that 'the value of y was always x'--the value 1 is substituted
+                     * for x when evaluating the gate identity.
+                     */
+                    if (true_when_y_val_zero) {
+                        // constraint: 0*x + 1*y + 0*0 + 0 == 0
+                        composer.create_add_gate({ .a = x.witness_index,
+                                                   .b = y.witness_index,
+                                                   .c = composer.zero_idx,
+                                                   .a_scaling = 0,
+                                                   .b_scaling = 1,
+                                                   .c_scaling = 0,
+                                                   .const_scaling = 0 });
+                        expected_result = false;
+                    } else {
+                        // constraint: 0*x + 1*y + 0*0 - 1 == 0
+                        composer.create_add_gate({ .a = x.witness_index,
+                                                   .b = y.witness_index,
+                                                   .c = composer.zero_idx,
+                                                   .a_scaling = 0,
+                                                   .b_scaling = 1,
+                                                   .c_scaling = 0,
+                                                   .const_scaling = -1 });
+
+                        expected_result = true;
+                    }
                 }
-            }
 
-            x.assert_equal(y);
+                x.assert_equal(y);
 
-            // both field elements have real value 1 now
-            EXPECT_EQ(x.get_value(), 1);
-            EXPECT_EQ(y.get_value(), 1);
+                // both field elements have real value 1 now
+                EXPECT_EQ(x.get_value(), 1);
+                EXPECT_EQ(y.get_value(), 1);
 
-            bool result = composer.check_circuit();
+                bool result = composer.check_circuit();
 
-            EXPECT_EQ(result, expected_result);
-        };
+                EXPECT_EQ(result, expected_result);
+            };
 
-        run_test(false);
-        run_test(true, true);
-        run_test(true, false);
+            run_test(false);
+            run_test(true, true);
+            run_test(true, false);
+        }
     }
 
-    static void test_add_mul_with_constants()
-    {
-        Composer composer = Composer();
-        auto gates_before = composer.get_num_gates();
-        uint64_t expected = fidget(composer);
-        auto gates_after = composer.get_num_gates();
-        EXPECT_EQ(composer.get_variable(composer.w_o[gates_after - 1]), fr(expected));
-        info("Number of gates added", gates_after - gates_before);
-        bool result = composer.check_circuit();
-        EXPECT_EQ(result, true);
-    }
```

Comment:
This isnt' really a test of field logic, rather it's a test of circuit builder logic.